marimo/_plugins/stateless/video.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2023 Marimo. All rights reserved.
+from __future__ import annotations
+
+import io
+from typing import Any, Optional, Union
+
+from marimo._output.builder import h
+from marimo._output.hypertext import Html
+from marimo._output.rich_help import mddoc
+from marimo._output.utils import create_style
+from marimo._plugins.core.media import io_to_data_url
+
+
+@mddoc
+def video(
+    src: Union[str, bytes, io.BytesIO, io.BufferedReader],
+    controls: bool = True,
+    muted: bool = False,
+    autoplay: bool = False,
+    width: Optional[int] = None,
+    height: Optional[int] = None,
+    rounded: bool = False,
+    style: Optional[dict[str, Any]] = None,
+) -> Html:
+    """Render an video as HTML.
+
+    **Example.**
+
+    ```python3
+    mo.video(
+        src="https://www.youtube.com/watch?v=dQw4w9WgXcQ",
```

Comment:
lol.

Youtube videos don't work with `<video>` tags though, you have to embed them in an iframe and change `watch?v=` to `embed/` in the URL

- Could automagically handle youtube videos. For the user that'd be nice. But there's a risk that YouTube changes their URL structure, embedding policy, etc.
- Could raise a warning if URL is for YouTube (since that'll be a common use case).
- Could just fail silently (like right now)

Reply:
i dont think we should do any magic. i just changed the src